core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -494,6 +495,19 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
           }
         }
         .flatMap { entry => EventLogFileReader(fs, entry) }
+        .filter { reader =>
+          try {
+            reader.modificationTime
+            true
+          } catch {
+            case e: Exception =>
```

Comment:
Could you use more specific exception instead of `java.io.Exception`?

Reply:
Changed to be more specific. Thanks.

Reply:
Changed to be more specific. Thanks.